Readme update (#23)
fossasia/badgemagic-app: README.md

- [FOSSASIA Shop here](https://fossasia.com/product/led-badge/)

## Reverse-Engineering Bluetooth LE Devices

Security in Bluetooth LE devices is optional, and many cheap products you can find on the market are not secured at all. This applies to our Bluetooth LED Badge. While this could lead to some privacy issues, this can also be a source of fun, especially when you want to use an LED Badge in a different way. It also makes it easy for us to get started with the development of a Free and Open Source ios app.

As we understand how the Bluetooth LED badge works, converting a text to multiple byte arrays, we can send using the Bluetooth LE APIs. An indepth blog post about reverse-engineering the Bluetooth community [is here](http://nilhcem.com/iot/reverse-engineering-bluetooth-led-name-badge).

The implementation in the ios app consists of manipulating bits. That may be tricky. A single bit error and nothing will work, plus it will be hard to debug. For those reasons, and since the specs are perfectly clear we recommend to start writing unit tests before the code implementation.

## Running / Development

> To run the project on iOS you need installed XCode and have a phisical device.

Reviewer: ```suggestion
To run the project on iOS you need installed XCode and have a physical device since simulators don't support bluetooth.
```

> - Generate the native part of the app using the command:

Reviewer: ```suggestion
- Build the app using. following commad: (It will generate `ios` directory containing native part of the app)
```

  ```bash
    yarn ios --configuration=Debug --device
  ```

> - Open `BadgeMagic.xcworkspace` in Xcode, then in the 'Signing and Capabilities' section, set up Development Team and Bundle Identifier.

Reviewer: ```suggestion
- Using Xcode open `BadgeMagic.xcworkspace` located inside generated `ios` directory, then in the 'Signing and Capabilities' section, set up Development Team (You personal one) and Bundle Identifier (you can add suffix with your name - it serves only for local development).
```

> - Check if the physical build destination is set up on the physical device (this may be useful to remove Push Notifications). If you have a different Apple ID between Xcode and your device, you need to allow the device to accept the Apple ID in Settings -> Device Management.

Reviewer: ```suggestion
- Check if the physical build destination is set up on the physical device (it may be necessary to remove Push Notifications).
```
